libs/jitsu-js/src/analytics-plugin.ts

```diff
@@ -410,19 +410,32 @@ function adjustPayload(
   s2s: boolean
 ): AnalyticsClientEvent {
   const runtime: RuntimeFacade = config.runtime || (isInBrowser() ? windowRuntime(config) : emptyRuntime(config));
-  const url = runtime.pageUrl();
+  const properties = payload.properties || {};
+  const url = properties.url || runtime.pageUrl();
   const parsedUrl = safeCall(() => new URL(url), undefined);
   const query = parsedUrl ? parseQuery(parsedUrl.search) : {};
-  const properties = payload.properties || {};
+  const referrer = runtime.referrer();
 
-  if (payload.type === "page" && url) {
+  if (properties.url) {
     properties.url = url.replace(hashRegex, "");
-    properties.path = fixPath(urlPath(url));
+  }
+  const ctxPage = {
+    path: url ? fixPath(urlPath(url)) : properties.path,
```

Comment:
wonder if this should prioritize the path provided through `properties` as well:

```js
path: properties.path || fixPath(urlPath(url))
```